src/main/java/uk/ac/cam/cl/dtg/teaching/pottery/repo/RepoFactory.java

```diff
@@ -101,6 +118,16 @@ public Repo createInstance(
     }
   }
 
+  public void initialiseInstance(String repoId, TaskCopy c, int validityMinutes)
+      throws RepoStorageException, RepoExpiredException, RepoNotFoundException {
+    Repo repo = getInstance(repoId, true);
+    RepoInfoWithStatus info = repo.toRepoInfo();
+    if (!info.isRemote()) {
```

Comment:
Could you put a todo in here that we need to deal with remote repos here?

Reply:
I can add a todo, but what is the difference between this and where we are at with remote repos at the moment (where they also don't get a skeleton)?

Reply:
Well at the moment they can get a skeleton by requesting the list of skeleton files and copying them in themselves. (That's what I do at the moment). But if in future you need to generate the skeleton files programmatically then this is not going to work. So I think we would actually need to clone the remote repo, write the files in and then push it back, and bork if we get a merge conflict.

Reply:
I see, that makes sense. I'll add a TODO